### PR TITLE
Update IO operator documentation

### PR DESCRIPTION
**Pull request by benharsh**
This PR updates the IO operator documentation in the modules to point to alternative methods on channels. This PR also updates the spec to warn that the IO operator is now deprecated.

Testing:
- [x] make docs

### CODE REVIEW

**Comment by benharsh** [comment]
Screen Shot 2022-09-07 at 11 21 11 AM

Here is a sample rendering of the warning.

File: doc/rst/language/spec/statements.rst [hunk 1]
```diff
@@ -267,6 +267,10 @@ the right-hand-side; or reading from the left-hand-side and storing the
 result in the variable on the right-hand-side. This operator can be
 chained with other I/O operator calls.
 
+.. warning::
+
+   The I/O operator is deprecated.
```
Comment (DanilaFe) [inline doc/rst/language/spec/statements.rst:272]: Other places in the docs seem to use a "note" for saying that the operator is deprecated: 
Screen Shot 2022-09-07 at 11 55 33 AM
Does the consistency matter at all?

File: modules/standard/IO.chpl [hunk 2]
```diff
@@ -3857,6 +3857,14 @@ proc channel.writeIt(const x) throws {
      on the type of channel. This operator returns a const reference to the
      same channel so that multiple operator calls can be chained together.
 
+     .. warning:: The <~> operator on channels is deprecated. The intended
+       alternative is to use methods like :proc:`channel.read` and
+       :proc:`channel.write` instead.
+
+       Methods that provide both reading and writing functionality will soon be
```
Comment (DanilaFe) [inline modules/standard/IO.chpl:3864]: As an end user, this is not clear to me. Does the IO operator provide both reading and writing functionality? Is it a method? Maybe a more general term than "method" would work if you're trying to explain why `<~>` is deprecated.

**Review by DanilaFe — APPROVED** [review]

File: doc/rst/language/spec/statements.rst (hunk 1 above)
Comment (benharsh) [reply doc/rst/language/spec/statements.rst:272]: Good catch, I'll update that in a moment.

File: modules/standard/IO.chpl (hunk 2 above)
Comment (benharsh) [reply modules/standard/IO.chpl:3864]: The IO operator does provide reading and writing functionality. My intention with this second paragraph was to point out that this general read-or-write functionality was going away, and so there wouldn't be a method to replace the <~> operator directly.

**Pull request merged**